src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -642,12 +641,12 @@ protected override HostObjectInitializationStatus InitializeHostObject()
 
                 // NOTE: For compat reasons this must remain ICscHostObject
                 // we can dynamically test for smarter interfaces later..
-                using (RCWForCurrentContext<ICscHostObject> hostObject = new RCWForCurrentContext<ICscHostObject>(HostObject as ICscHostObject))
+                if (HostObject is ICscHostObject hostObjectCOM)
                 {
-                    ICscHostObject cscHostObject = hostObject.RCW;
-
-                    if (cscHostObject != null)
+                    using (RCWForCurrentContext<ICscHostObject> hostObject = new RCWForCurrentContext<ICscHostObject>(hostObjectCOM))
                     {
+                        ICscHostObject cscHostObject = hostObject.RCW;
```

Comment:
Just checked, no - regardless of COM code, the getter for `RCW` throws if the underlying field is null. http://source.roslyn.io/#Microsoft.Build.Tasks.CodeAnalysis/RCWForCurrentContext.cs,81
(The constructor, where the COM stuff is, seems to also assert that it's not null after COM calls, using... the original object (???) if the calls fail)